Navigation component

TakoLeakest opened the pull request:
- Adds main component to navigate through all the different pages
- Moves ParticleJS to its own component (To be rendered for all pages in _app.tsx)

Closes #20

Commit:
Adds background and text for home

- Adds name, navigation bar, and page main text to screen
- Requires CSS to set positions

Commit:
Adds basic navigation to navbar

- Adds Link to NavigationItem for routing
- Removing Navbar from _app (Changes title with each page)
- Modifies index of pages to use Navbar to change its titles correctly

Commit:
Adds SVG styles for navigation

- Modifies simple divs to use SVGs for the non-square shapes
- Deletes images since they were also replaced by SVGs
- Modifies SCSS to adjust for these changes and also importing K2D font

Commit:
Adds movement of circle based on page

- Adds CircleX enum to get the X positiion of the circle based on the page
- Modifies the parameters passed to Navigation to select the proper x for circle

Commit:
Merge branch 'main' of https://github.com/TakoLeakest/kaichou-graduation

Commit:
Merge branch 'main' of https://github.com/TakoLeakest/kaichou-graduation

Commit:
Transforms Particle in a single component

- Moves the code for the Particles to a single component (Also moving its SCSS)
- Modifies _app to always render the SakuraParticles component
- Deletes sakura-tree image used for testing, waiting for another one with permission

TakoLeakest commented:
*Results from this pull request:*

![navigation](https://user-images.githubusercontent.com/85906956/122311977-f09cd580-ced8-11eb-81eb-50f4ba518381.png)

RedMaps commented on packages/frontend/components/Navigation.tsx:
```diff
@@ -0,0 +1,98 @@
+                    <path d="M0 -20H650V88.8983L145.013 106L0 -20Z" fill="#FF7B39" />
+                </g>   
+                <g>
+                    <NavigationItem
+                        href="/"
+                        x="189"
+                        y="26"
+                        width="38"
+                        height="40"
+                        path="M204.984 26.5735C205.829 25.8637 206.897 25.4746 208 25.4746C209.103 25.4746 210.171 25.8637 211.016 26.5735L225.078 38.3985C225.602 38.8383 226.022 39.3874 226.311 40.0071C226.6 40.6268 226.75 41.3022 226.75 41.986V61.6266C226.75 62.4554 226.421 63.2502 225.835 63.8363C225.249 64.4223 224.454 64.7516 223.625 64.7516H215.031C214.202 64.7516 213.408 64.4223 212.822 63 ...
```
Do you think you could make these into actual react components with .svg file imports in front of the svg? It would probably make it easier to move stuff around and reduce the clutter of having the entire svg path in there. Also helps with accessibility. If you don't want to or can't ill do that eventually so don't worry about it ^^

TakoLeakest replied:
Ok, I'll try that

TakoLeakest replied:
This is the best I could do, is it good? Couldn't get to import the svgs, but this is more manageable (It can be moved around, resized, etc)

![image](https://user-images.githubusercontent.com/85906956/122411299-43fb3c00-cf4a-11eb-9d9a-f803a24f0321.png)

![image](https://user-images.githubusercontent.com/85906956/122411171-27f79a80-cf4a-11eb-9a53-6a233fe61bfb.png)

Commit:
Refactoring Navigation Items

- Modifying Navigation items to just be nested SVGs, allowing for easier control
- Adds sakura trees image (With permission)

Commit:
Merge branch 'main' of https://github.com/Kaichou-Project/kaichou-graduation

Commit:
Modifies visualization of SVGs and Particles (Quick Fix)

- Changes max-width to width for the SVGs inside Navigation
- Moves SakuraParticles from app to Navbar
- Modifies SakuraParticles to only be inside Navigation component

TakoLeakest committed:
Merge branch 'Kaichou-Project:main' into main

Commit:
Applies prettier to Navigation components (Quick Fix)

- Modifies tsx files created by TakoLeakest to follow prettier standar

Commit:
Fixes missing prettier format in file (Quick Fix)

- Modifies NavigationList formatting to work with prettier
- Modifies tab space in Navigation SCSS

RedMaps reviewed (approved):
![EnGVR7mXYAAuYBS](https://user-images.githubusercontent.com/16868419/122464985-c66e1500-cfb7-11eb-8037-9a7ae164d5eb.jpg)
